GDJS/Runtime/layer.ts

```diff
@@ -287,18 +320,54 @@ namespace gdjs {
       const sinValue = Math.sin(angleInRadians);
       x = cosValue * x - sinValue * y;
       y = sinValue * tmp + cosValue * y;
-      return [x + this.getCameraX(cameraId), y + this.getCameraY(cameraId)];
+      result[0] = x + this.getCameraX(cameraId);
+      result[1] = y + this.getCameraY(cameraId);
+
+      return result;
     }
 
     /**
-     * Convert a point from the scene coordinates (for example,
+     * Convert a point from the container coordinates (for example,
      * an object position) to the canvas coordinates.
      *
-     * @param x The x position, in scene coordinates.
-     * @param y The y position, in scene coordinates.
+     * @param x The x position, in container coordinates.
+     * @param y The y position, in container coordinates.
      * @param cameraId The camera number. Currently ignored.
-     */
-    convertInverseCoords(x: float, y: float, cameraId?: integer): FloatPoint {
+     * @param result The point instance that is used to return the result.
+     */
+    convertInverseCoords(
+      x: float,
+      y: float,
+      cameraId: integer = 0,
+      result: FloatPoint
+    ): FloatPoint {
+      let position = result || [0, 0];
+      // TODO EBO use an AffineTransformation to avoid chained calls.
+      this.applyLayerTransformation(x, y, cameraId, position);
+      return this._runtimeScene.convertInverseCoords(
```

Comment:
Can you explain why these two transformation and usage of this._runtimeScene here?

Reply:
`this._runtimeScene` is an `InstanceContainer` but I kept the old name for compatibility with community extensions (the embedded extension were using it so I was afraid to break something)

I think there is an issue there, I used the same method for visual transformations and hitboxes transformations. I'll redo this and write tests.

Reply:
Actually, it works fine. I added some tests.